.gitattributes

```diff
@@ -0,0 +1,43 @@
+* text eol=lf
+
+#
+# The above will handle all files NOT found below
+# https://help.github.com/articles/dealing-with-line-endings/
+# https://github.com/Danimoth/gitattributes
+
+# These are explicitly windows files and should use crlf
+*.bat           text eol=crlf
+
+# These files are text and should be normalized (Convert crlf => lf)
+*.bash          text eol=lf
+*.css           text diff=css
+*.htm           text diff=html
+*.html          text diff=html
+*.java          text diff=java
```

Comment:
If I now check out on Windows, do I now get CRLF or LF?

Reply:
From the documentation:

> When more than one pattern matches the path, a later line overrides an earlier line. This overriding is *done per attribute*.
> \- https://git-scm.com/docs/gitattributes

This means that the top line `* text eol=lf` sets the `eol` attribute for all files `*`. The line `*.java          text diff=java` sets the `diff` attribute to `java` while leaving the `eol` attribute alone.